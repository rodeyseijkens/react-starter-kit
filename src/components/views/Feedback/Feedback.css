@import '../../globals/variables.css';

.root {
  background: #f5f5f5;
  color: var(--default-font-color);
  font-family: var(--default-font-family);
}

.container {
  font-size: 1.5rem; /* ~24px */
  margin: 0 auto;
  max-width: var(--screen-xl);
  padding: var(--default-padding-large) var(--default-padding-small);
  text-align: center;
}

.link,
.link:active,
.link:hover,
.link:visited {
  color: var(--default-font-color);
  text-decoration: none;
}

.link:hover {
  text-decoration: underline;
}

.spacer {
  padding-left: var(--default-padding);
  padding-right: var(--default-padding);
}
