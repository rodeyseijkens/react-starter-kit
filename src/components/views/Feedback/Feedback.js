import React, { PureComponent } from 'react';
import themableWithStyles from 'themableWithStyles';

import styles from './Feedback.css';

@themableWithStyles(styles)
class Feedback extends PureComponent {
  render() {
    return (
      <section className={styles.root}>
        <div className={styles.container}>
          <a
            target="_blank"
            rel="noopener noreferrer"
            className={styles.link}
            href="https://gitter.im/kriasoft/react-starter-kit"
          >
            Ask a question
          </a>
          <span className={styles.spacer}>|</span>
          <a
            target="_blank"
            rel="noopener noreferrer"
            className={styles.link}
            href="https://github.com/kriasoft/react-starter-kit/issues/new"
          >
            Report an issue
          </a>
        </div>
      </section>
    );
  }
}

export default Feedback;
