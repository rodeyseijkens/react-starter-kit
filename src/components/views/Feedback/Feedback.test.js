/* eslint-env jest */
/* eslint-disable padded-blocks, no-unused-expressions */

import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Feedback from './Feedback';

describe('<Feedback />', () => {
  const wrapper = shallow(<Feedback.ComposedComponent />);

  it('renders children correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
