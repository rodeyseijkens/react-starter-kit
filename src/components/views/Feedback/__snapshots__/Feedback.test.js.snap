// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Feedback /> renders children correctly 1`] = `
<section
  className="root"
>
  <div
    className="container"
  >
    <a
      className="link"
      href="https://gitter.im/kriasoft/react-starter-kit"
      rel="noopener noreferrer"
      target="_blank"
    >
      Ask a question
    </a>
    <span
      className="spacer"
    >
      |
    </span>
    <a
      className="link"
      href="https://github.com/kriasoft/react-starter-kit/issues/new"
      rel="noopener noreferrer"
      target="_blank"
    >
      Report an issue
    </a>
  </div>
</section>
`;
